### PR TITLE
Feat/rl-7/weather-investigation-init

### PR DESCRIPTION
**Pull request by ahureikinandrei**
weather_data
weather_on_query
error_query

### CODE REVIEW

File: client/src/api/UserService.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    static async createUser(
+        email: string,
+        password: string
+    ): Promise<AxiosResponse> {
+        const body = {
+            email,
+            password,
+        }
+
+        return axios.post('http://localhost:5000/api/user', body)
```
Comment (aadereiko) [inline client/src/api/UserService.ts]: `http://localhost:5000` url shouldn't be written manually

File: client/src/api/GeoLocationService.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { IGeolocationResponse } from './types'
+
+export default class GeoLocationService {
+    static async getCurrentLocationCoords(): Promise<IGeolocationResponse> {
```
Comment (aadereiko) [inline client/src/api/GeoLocationService.ts]: you don't need to do it `async` if it doesn't use `await`

File: client/src/api/UserService.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { AxiosResponse } from 'axios'
+import axios from '../core/axios'
+
+export default class UserService {
+    static async getUsers(): Promise<AxiosResponse> {
```
Comment (aadereiko) [inline client/src/api/UserService.ts]: remove an useless word async

File: client/src/api/AuthService.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import { AxiosResponse } from 'axios'
+import axios from '../core/axios'
+
+export default class AuthService {
+    static async validateUserToken(): Promise<AxiosResponse> {
```
Comment (aadereiko) [inline client/src/api/AuthService.ts]: an useless async

File: client/src/components/AuthModal/AuthModal.tsx [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+                                label="Sign Up"
+                            />
+                        </Tabs>
+                    </div>
+                    <IconButton
+                        onClick={onClose}
+                        color="secondary"
+                        aria-label="close"
+                    >
+                        <CloseIcon style={{ fontSize: 26 }} color="secondary" />
```
Comment (aadereiko) [inline client/src/components/AuthModal/AuthModal.tsx]: try to avoid inline styles

File: client/src/components/AuthModal/AuthModal.tsx [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+                    index={AuthModalItemsValue.SIGN_IN}
+                    /* eslint-disable-next-line react/jsx-props-no-spreading */
+                    {...a11yProps(AuthModalItemsValue.SIGN_IN)}
+                >
+                    <SignInTabPanel onClose={onClose} />
+                </TabPanel>
+                <TabPanel
+                    value={value}
+                    index={AuthModalItemsValue.SIGN_UP}
+                    /* eslint-disable-next-line react/jsx-props-no-spreading */
```
Comment (aadereiko) [inline client/src/components/AuthModal/AuthModal.tsx]: put this eslint rule to a `.eslintrc` file

File: client/src/components/HeaderSettings/HeaderSettings.tsx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+const HeaderSettings: FC = () => {
+    const classes = useStylesHeader()
+    const isAuth = useTypedSelector(selectAuthStatus)
+    const isLoading = useTypedSelector(selectAuthIsLoading)
+    const { logout } = useActions()
+    const [visibleModal, setVisibleModal] = useState(false)
+
+    const handleClickOpenModal = (): void => {
+        if (isLoading) {
+            return
```
Comment (aadereiko) [inline client/src/components/HeaderSettings/HeaderSettings.tsx]: it's better to disable button while it's waiting

**Review by aadereiko — CHANGES_REQUESTED** [review]
really nice job :)
some small things, when you fix them, just merge it

File: client/src/api/UserService.ts (hunk 1 above)
Comment (ahureikinandrei) [reply client/src/api/UserService.ts]: corrected

File: client/src/api/GeoLocationService.ts (hunk 2 above)
Comment (ahureikinandrei) [reply client/src/api/GeoLocationService.ts]: corrected

File: client/src/api/AuthService.ts (hunk 4 above)
Comment (ahureikinandrei) [reply client/src/api/AuthService.ts]: corrected

File: client/src/components/AuthModal/AuthModal.tsx (hunk 5 above)
Comment (ahureikinandrei) [reply client/src/components/AuthModal/AuthModal.tsx]: corrected

File: client/src/components/AuthModal/AuthModal.tsx (hunk 6 above)
Comment (ahureikinandrei) [reply client/src/components/AuthModal/AuthModal.tsx]: corrected

File: client/src/components/HeaderSettings/HeaderSettings.tsx (hunk 7 above)
Comment (ahureikinandrei) [reply client/src/components/HeaderSettings/HeaderSettings.tsx]: corrected

**Pull request merged**